src/Sylius/Bundle/CoreBundle/Migrations/Version20240315112656.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Sylius\Bundle\CoreBundle\Doctrine\Migrations\AbstractMigration;
+
+final class Version20240315112656 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Replace DC2TYPE:array with JSON';
+    }
+
+    public function up(Schema $schema): void
+    {
+        foreach ($this->tablesAndColumnsToBeUpdated() as [$table, $column]) {
+            $this->changeTypesFromLongtextToJsonAndEncodeSerializedData($table, $column);
+        }
+    }
+
+    public function down(Schema $schema): void
+    {
+        foreach ($this->tablesAndColumnsToBeUpdated() as [$table, $column]) {
+            $this->changeTypesFromJsonToLongtext($table, $column);
+        }
+    }
+
+    public function postDown(Schema $schema): void
+    {
+        parent::postDown($schema);
+
+        foreach ($this->tablesAndColumnsToBeUpdated() as [$table, $column]) {
+            $this->serializeEncodedData($column, $table);
```

Comment:
Method call doesn't match method's signature.
```php
private function serializeEncodedData(string $table, string $dataColumn): void
```

```suggestion
            $this->serializeEncodedData($table, $column);
```

Reply:
Thank you for that catch, that would be horrible 😅